packages/git/src/browser/git-quick-open-service.ts

```diff
@@ -343,6 +343,94 @@ export class GitQuickOpenService {
+            onType(lookFor: string, acceptor: (items: QuickOpenItem[]) => void): void {
+                const dynamicItems: QuickOpenItem[] = [];
+                const suffix = "Press 'Enter' to confirm or 'Escape' to cancel.";
+
+                if (lookFor === undefined || lookFor.length === 0) {
+                    dynamicItems.push(new SingleStringInputOpenItem(
+                        `Stash changes. ${suffix}`,
+                        () => doStash(lookFor)
+                    ));
+                } else {
+                    dynamicItems.push(new SingleStringInputOpenItem(
+                        `Stash changes with message: ${lookFor}. ${suffix}`,
+                        () => doStash(lookFor)
+                    ));
+                }
+                acceptor(dynamicItems);
+            }
+        };
+        this.quickOpenService.open(quickOpenModel, this.getOptions('Stash message', false));
+    }
+
+    protected async doStashAction(action: 'pop' | 'apply' | 'drop', text: string, getMessage?: () => Promise<string>) {
+        const repo = this.repositoryProvider.selectedRepository;
+        if (repo) {
+            const list = await this.git.stash(repo, { action: 'list' });
+            if (list) {
+                const quickOpenItems = list.map(stash => new GitQuickOpenItem<Git.StashEntry>(stash, async () => {
+                    try {
+                        await this.git.stash(repo, {
+                            action,
+                            id: stash.id
+                        });
+                        if (getMessage) {
+                            this.messageService.info(await getMessage());
+                        }
+                    } catch (error) {
+                        this.gitErrorHandler.handleError(error);
+                    }
+                }, () => stash.message));
+                this.open(quickOpenItems, text);
+            }
+        }
+    }
+
+    async applyStash(pop?: boolean): Promise<void> {
+        this.doStashAction(pop ? 'pop' : 'apply', `Select a stash to ${pop ? 'pop' : 'apply'}.`);
+    }
+
+    async dropStash(): Promise<void> {
+        this.doStashAction('drop', 'Select a stash entry to remove it from the list of stash entries.',
+            async () => {
+                if (this.repositoryProvider.selectedRepository) {
+                    const list = await this.git.stash(this.repositoryProvider.selectedRepository, { action: 'list' });
+                    const stashCount = list ? list.length : 0;
+                    let listString = '';
+                    if (list) {
+                        list.forEach(stashEntry => {
+                            listString += stashEntry.message + '\n';
+                        });
+                    }
+                    return `Stash successfully removed.
+                    There ${stashCount === 1 ? 'is' : 'are'} ${stashCount || 'no'} more entry in stash list.
+                    \n${listString}`;
```

Comment:
I am not sure if we need this detailed info after dropping from the stash.

![screen shot 2019-02-26 at 16 25 57](https://user-images.githubusercontent.com/1405703/53424367-594ba580-39e3-11e9-9c20-37aa67c04c47.jpg)

Reply:
I wanted to have at least something because this would be the only action without any feedback which I as a user find a bit confusing.

Reply:
OK, let's leave as is. If we decide to remove it later, the users can still do `git stash list`.